src/App.tsx

```diff
@@ -1,61 +1,108 @@
+import { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+const titleInputId = 'title';
+const userInputId = 'user';
+const userSelectDefaultValue = '0';
+
+const allTodos = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(user => user.id === todo.userId),
+}));
 
 export const App = () => {
+  const [isTitleEmpty, setIsTitleEmpty] = useState(false);
+  const [isUserEmpty, setIsUserEmpty] = useState(false);
+  const [maxTodoId, setMaxTodoId] = useState(
+    Math.max(...allTodos.map(todo => todo.id)),
+  );
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form
+        action="/api/todos"
+        method="POST"
+        onSubmit={event => {
+          event.preventDefault();
+
+          const titleInput
+            = document.getElementById(titleInputId) as HTMLInputElement;
```

Comment:
why do you use getElementById?
add state with title and set title in input onChange

Reply:
I need values from input only when user clicks `Add` button. I think, save values every time when user edits them, is very bad and redundant.